edge-modules/TestAnalyzer/TestResultAnalysis.cs

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace TestAnalyzer
+{
+    using System.Collections.Generic;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Serialization;
+
+    [JsonObject(NamingStrategyType = typeof(CamelCaseNamingStrategy))]
+    class TestResultAnalysis
+    {
+        public TestResultAnalysis(IList<ModuleMessagesReport> messagesReport, IList<ResponseOrientedReport> dmReport, IList<ResponseOrientedReport> twinsReport)
+        {
+            this.MessagesReport = messagesReport;
+            this.DmReport = dmReport;
+            this.TwinsReport = twinsReport;
+        }
+
+        public IList<ResponseOrientedReport> DmReport { get; }
+        public IList<ResponseOrientedReport> TwinsReport { get; }
+        public IList<ModuleMessagesReport> MessagesReport { get; }
+
+        public override string ToString()
+        {
+            return JsonConvert.SerializeObject(this, Formatting.Indented);
```

Comment:
Are we going to change this to implement it explicitly for each property? if so, please add a todo here.

Reply:
ModuleMessagesReport is serializable, however what we are formatting is a list of ModuleMessagesReports. So the wrapping list is what requires serialization here.

If I'm understanding your question correctly, we already do implement the serialization explicitly in ModuleMessagesReport. I can add a comment explaining this though since it is not straightforward.